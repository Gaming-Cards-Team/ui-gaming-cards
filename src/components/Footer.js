import React from "react";
import Navbar from "react-bootstrap/Navbar";
import "bootstrap/dist/css/bootstrap.min.css";
import { FOOTER_TEXT } from "../util/constant.js"

const textStyle={
    color: "white",
    width: "100%",
    textAlign: "center",
  };
  const footerStyle={
    backgroundColor: "#2d963f"
  };

const Footer = () => (
  <Navbar 
  fixed="bottom"
  style={footerStyle}
  >
    <div
      data-testid="footer-text"
      style={textStyle}
    >
      {FOOTER_TEXT}
    </div>
  </Navbar>
);

export default Footer;
