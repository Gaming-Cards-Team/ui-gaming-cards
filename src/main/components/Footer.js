import React from "react";
import Navbar from "react-bootstrap/Navbar";
import "bootstrap/dist/css/bootstrap.min.css";

const Footer = () => (
  <Navbar bg="success" fixed="bottom">
    <div
      data-testid="footer-text"
      style={{
        color: "white",
        width: "100%",
        textAlign: "center",
      }}
    >
      Star Wars 2020 All rights reserved. Copyright Level Upâ€©
    </div>
  </Navbar>
);

export default Footer;
