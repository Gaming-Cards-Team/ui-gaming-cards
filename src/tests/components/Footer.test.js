import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Footer from '../../main/components/Footer';

describe('Test Footer component', () => {

    it('should have a fixed text', () => {
        
        const { getByTestId } = render(<Footer />);
        const footerText = getByTestId('footer-text')

        expect (footerText.textContent).toBe('Star Wars 2020 All rights reserved. Copyright Level Upâ€©')
    });


    it('should validate the Footer component', () => {
        
        const footer = renderer.create(<Footer />).toJSON();
        expect(footer).toMatchSnapshot();
    });

});