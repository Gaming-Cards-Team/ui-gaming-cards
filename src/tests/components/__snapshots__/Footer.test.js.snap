// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer component should validate the Footer component 1`] = `
<nav
  className="navbar navbar-expand navbar-light bg-success fixed-bottom"
>
  <div
    data-testid="footer-text"
    style={
      Object {
        "color": "white",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    Star Wars 2020 All rights reserved. Copyright Level Upâ€©
  </div>
</nav>
`;
